src/ex1/script.js

```diff
@@ -0,0 +1,183 @@
+    setSortButtonDisplay("flex");
+    setEmptyTasksDivDisplay("none");
+    setBottomDivBarDisplay("flex");
+    setTasksDivContainerDisplay("flex");
+  }
+  injectNewTaskToHtml(text);
+}
+
+function setDispalyToEmptyTasks() {
+  setSortButtonDisplay("none");
+  setEmptyTasksDivDisplay("block");
+  setBottomDivBarDisplay("none");
+  setTasksDivContainerDisplay("none");
+}
+
+function firstCharIsNotCapital(text) {
+  return text[0].toUpperCase() !== text[0];
+}
+
+function capitalizeFirstLetter(text) {
+  const capitalized = text.charAt(0).toUpperCase() + text.slice(1);
+  return capitalized;
+}
+
+function onClickClearAll() {
+  clearTasksHtml();
+  resetTaskList();
+}
+
+function clearTasksHtml() {
+  const tasks_list_div = document.getElementById("tasks_list_container");
+  tasks_list_div.innerHTML = "";
+}
+function resetTaskList() {
+  tasks_list = [];
+  tasks_amount = 0;
+  setDispalyToEmptyTasks();
+}
+
+function setBottomDivBarDisplay(displayStatus) {
+  const buttom_div_bar = document.getElementById("buttom_bar_container");
+  buttom_div_bar.style.display = displayStatus;
+}
+function setTasksDivContainerDisplay(displayStatus) {
+  const tasks_list_div = document.getElementById("tasks_list_container");
+  tasks_list_div.style.display = displayStatus;
+}
+function setEmptyTasksDivDisplay(displayStatus) {
+  const tasks_list_div = document.getElementById("empty_tasks_text");
+  tasks_list_div.style.display = displayStatus;
+}
+function setSortButtonDisplay(displayStatus) {
+  const sort_display = document.getElementById("sort");
+  sort_display.style.display = displayStatus;
+}
+
+function updatePendingTask() {
+  const pending_tasks_p_tag = document.getElementById(
+    "amount_of_pending_tasks"
+  );
+  pending_tasks_p_tag.innerText = "You have " + tasks_amount + " pending tasks";
+}
+
+function sortTasksList() {
+  clearTasksHtml();
+  const sort_arrow = document.getElementById("sort_arrow");
+  if (sort_direction === "down") {
+    tasks_list.sort((a, b) => {
+      return a.text.toLowerCase().localeCompare(b.text.toLowerCase());
+    });
+    sort_direction = "up";
+
+    sort_arrow.src = arrow_down_src;
+  } else {
+    tasks_list.sort((a, b) => {
+      return b.text.toLowerCase().localeCompare(a.text.toLowerCase());
+    });
+    sort_direction = "down";
+    sort_arrow.src = arrow_up_src;
+  }
+
+  const task_list_div = document.getElementById("tasks_list_container");
+  tasks_list.forEach((object) => {
+    task_list_div.appendChild(object.element);
+  });
+}
+
+function removeTaskFromList(task) {
+  const index = tasks_list.findIndex((object) => {
+    return object.text === task;
+  });
+
+  tasks_list.splice(index, 1);
+  updateTasksAmount("-");
+  if (tasks_amount === 0) {
+    setDispalyToEmptyTasks();
+  }
+  updatePendingTask();
+}
+
+function updateTasksAmount(action) {
+  switch (action) {
+    case "+": {
+      tasks_amount++;
+      break;
+    }
+    case "-": {
+      tasks_amount--;
+      break;
+    }
+  }
+}
```

Comment:
It's better practice to leave functions immutable, in this case, meaning that the function will get tasks_amount as a parameter, return tasks_amount +/- 1 and you will assign it to tasks_amount yourself. This way, the function can be extracted to an external util scripts file (just as an example).